### DISCUSSION TITLE
Give the contribution of each variant to the overall PGS

### DISCUSSION THREAD

**Discussion by CristianRiccio** [PGScatalog/pgsc_calc#131] [Ideas]
I would like to know the contribution (number) of each variant to the overall PGS for each individual.

Is this possible with PGS Calculator?

See my question on Bioinformatics Stackexchange: https://bioinformatics.stackexchange.com/questions/21367/can-the-pgs-calculator-give-the-contribution-of-each-variant-to-the-overall-pgs

**Comment by smlmbrt** [comment]
Hi @CristianRiccio, this information (per:variant effect_weight*dosage) isn't provided as those are intermediate data that are summed within plink. The effect sizes for each variant are provided in the `effect_weight` column of the variant matching log file (https://pgsc-calc.readthedocs.io/en/latest/output.html) named `[sampleset]_log.csv.gz`. Note that the scale of the weights will be different for different scores, and the variant importance is also a function of allele frequency.

**Reply by CristianRiccio** [reply]
Hello @smlmbrt and thank you for the prompt reply.
Would it be possible to have an option to write these intermediary files to disk?

**Reply by smlmbrt** [reply]
Not within the pipeline, as they're not really useful in the core functionality and it would take space/time. You can use the scoring files to get the variant IDs and then provide them to plink to export dosages (["--export Av"](https://www.cog-genomics.org/plink/2.0/data#export), https://www.cog-genomics.org/plink/2.0/formats#traw). You could then combine the variant dosages and the effect_weights in the scoring files to get the the type of data you're looking for.
